processing/src/main/java/io/druid/granularity/AllGranularity.java

```diff
@@ -62,7 +62,7 @@ public boolean equals(Object o)
   @Override
   public int hashCode()
   {
-    return 0;
+    return getClass().hashCode();
```

Comment:
otherwise All and None have a hash collision. Minor detail, and I can take it out if people are worried about the impact.